### ISSUE TITLE
Dot at end of folder name or file name gets stripped off.

### ISSUE THREAD

**Issue by kfogel** [PermanentOrg/sftp-service#69]
TL;DR: A dot at the end of an item name gets stripped off when we `rclone copy` down.  This happens for both files and folders.

I'm testing with my usual archive.  Below is the tree I expect to get when I run `rclone copy --create-empty-src-dirs permanent-prod:/ my-permanent-data` (note that I've appended "/" at the ends of folder names below, just to indicate that they are folders, but the "/" is not really there and is not actually part of the name of course).  The "###" comments show the four affected objects, two files and two folders:

```
archives/
  kfogel OTS test (13539)/
    My Files/
      dennys-fake-maple-syrup-2022-05-11.jpg
      file-ending-with-dot.             <--- ### trailing dot gets stripped off ###
      file-with-.-dot-in-middle-and-at-end.      <--- ### trailing dot gets stripped off ###
      Screenshot from 2022-07-12 14-17-51.png   
      tex-kerning.png   
      worker-digging.png
      Cables I have known/
        50-foot-VGA-cable.jpg
      GNOME Workspace Switcher bug/  
        gnome-3-shell-42.0-workspace-switcher-bottom-indicator-ws1-to-ws2-2022-04-10.png  
        gnome-3-shell-42.0-workspace-switcher-bottom-indicator-ws2-to-ws1-2022-04-10.png
        This subfolder is just to, you know, test subfolders./      <--- ### trailing dot gets stripped off ###
          uncle-neatest-whiskey-weird-endorsement.jpg
        This is an empty subfolder./      <--- ### trailing dot gets stripped off ###
```

The version of rclone probably doesn't matter for this bug, but just for completeness' sake here it is:

```
$ rclone --version
rclone v1.60.1-DEV
- os/version: debian bookworm/sid (64 bit)
- os/kernel: 6.1.0-1-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.19.4
- go/linking: dynamic
- go/tags: none
```

**Comment by Fenn-CS** [comment]
Thanks @kfogel, on this!

**Comment by Fenn-CS** [comment]
I looked into this with the local source of this repo and it appears the problem is from the [back-end](https://github.com/PermanentOrg/back-emd)

The names that are actual synced down are the `downloaodName`'s (Which appears to strip out end-trailing dots from display names)

**Comment by Fenn-CS** [comment]
In the case where there are multiple end-trailing dots in the `displayName` one is stripped off.


_`displayName` and `downloadName` view `Postgres`_
![excluded-dots](https://user-images.githubusercontent.com/14317775/216687650-b97d6e5d-7b87-4bf5-a5fa-0bfe8dc59144.png)

**Comment by kfogel** [comment]
@Fenn-CS In the code that sets downloadName originally, what does that code do when an item name ends in a single dot?

**Comment by kfogel** [comment]
@Fenn-CS Also, why are we stripping dots at all, even in the multiple-ending-dots case?

**Comment by Fenn-CS** [comment]
@kfogel It's a bug, I can't remember any reason why that's intentional although there was some kinda commit related to this in the backend still looking to see where it gets pulled out.

And yes it removes for single-end-trailing dots, I was just specifying that if there's more than one dot, the only the last one is removed so it's a specificity about the bug.

**Comment by kfogel** [comment]
@Fenn-CS Heh.  If I read "In the case where there are multiple end-trailing dots in the displayName one is stripped off", I'm naturally going to interpret that as something that happens only when there are multiple dots :-), because otherwise one would just say "If there are any trailing dots, exactly one is stripped off", which is an accurate description of the actual behavior.

Let's please find that commit and see if there was any reason for this, yeah...

**Comment by kfogel** [comment]
@Fenn-CS Another way to say it is: we don't know that it's the _last_ dot that gets removed -- we only know that there is one fewer dots than before :-).

**Comment by Fenn-CS** [comment]
> If I read "In the case where there are multiple end-trailing dots in the displayName one is stripped off", I'm naturally going to interpret that as something that happens only when there are multiple dots :-), because otherwise one would just say "If there are any trailing dots, exactly one is stripped off", which is an accurate description of the actual behavior.


That was a follow up comment. The issue is already well stated.

So, I was testing what happens if there are many dots at the end.

I found out, and added a follow up comment that describes what happens in that particular case.


FWIW, it's just an addition to thread.

> Another way to say it is: we don't know that it's the last dot that gets removed -- we only know that there is one fewer dots than before :-).


Actually, the end-trailing dot aka last dot is the one that is being removed.

**Comment by kfogel** [comment]
@Fenn-CS *nod* I understand what you're saying; that interpretation makes sense too, but for some reason my mind went the other way when I read it.  Thank you for clarifying what you meant.

Any luck finding the root cause?

**Comment by Fenn-CS** [comment]
@kfogel Yes, I managed to get hold of it last night. I just filed this back-end issue now to answer your question as well as serve its standard purpose of tracking the issue for discussion and eventual resolution.

https://github.com/PermanentOrg/back-end/issues/402

I would file in a fix for this issue today with perhaps a couple of follow up updates that maybe required to fix the issue every environment.
